service/operator/service.go

```diff
@@ -387,23 +402,25 @@ func (s *Service) GetDeleteState(obj, currentState, desiredState interface{}) (i
 		{
 			newData := map[string]string{}
 			for k, v := range deleteState.ConfigMap.Data {
-				if inConfigMapData(dState.ConfigMapData, k, v) {
+				if !inConfigMapData(dState.ConfigMapData, k, v) {
```

Comment:
I think it won't work in flannel-operator. At least it would be not easy to use. Let's say we have etcd entry to be removed. Then delete state should contain all entries but those to be removed, correct? If so then this makes `ProcessDeleteState` unnecessary complicated and error prone.